drivers/vmwarevsphere/flags.go

```diff
@@ -0,0 +1,252 @@
+)
+
+func (d *Driver) GetCreateFlags() []mcnflag.Flag {
+	return []mcnflag.Flag{
+		mcnflag.IntFlag{
+			EnvVar: "VSPHERE_CPU_COUNT",
+			Name:   "vmwarevsphere-cpu-count",
+			Usage:  "vSphere CPU number for docker VM",
+			Value:  defaultCpus,
+		},
+		mcnflag.IntFlag{
+			EnvVar: "VSPHERE_MEMORY_SIZE",
+			Name:   "vmwarevsphere-memory-size",
+			Usage:  "vSphere size of memory for docker VM (in MB)",
+			Value:  defaultMemory,
+		},
+		mcnflag.IntFlag{
+			EnvVar: "VSPHERE_DISK_SIZE",
+			Name:   "vmwarevsphere-disk-size",
+			Usage:  "vSphere size of disk for docker VM (in MB)",
+			Value:  defaultDiskSize,
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_BOOT2DOCKER_URL",
+			Name:   "vmwarevsphere-boot2docker-url",
+			Usage:  "vSphere URL for boot2docker image",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_VCENTER",
+			Name:   "vmwarevsphere-vcenter",
+			Usage:  "vSphere IP/hostname for vCenter",
+		},
+		mcnflag.IntFlag{
+			EnvVar: "VSPHERE_VCENTER_PORT",
+			Name:   "vmwarevsphere-vcenter-port",
+			Usage:  "vSphere Port for vCenter",
+			Value:  defaultSDKPort,
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_USERNAME",
+			Name:   "vmwarevsphere-username",
+			Usage:  "vSphere username",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_PASSWORD",
+			Name:   "vmwarevsphere-password",
+			Usage:  "vSphere password",
+		},
+		mcnflag.StringSliceFlag{
+			EnvVar: "VSPHERE_NETWORK",
+			Name:   "vmwarevsphere-network",
+			Usage:  "vSphere network where the docker VM will be attached",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_DATASTORE",
+			Name:   "vmwarevsphere-datastore",
+			Usage:  "vSphere datastore for docker VM",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_DATACENTER",
+			Name:   "vmwarevsphere-datacenter",
+			Usage:  "vSphere datacenter for docker VM",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_FOLDER",
+			Name:   "vmwarevsphere-folder",
+			Usage:  "vSphere folder for the docker VM. This folder must already exist in the datacenter",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_POOL",
+			Name:   "vmwarevsphere-pool",
+			Usage:  "vSphere resource pool for docker VM",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_HOSTSYSTEM",
+			Name:   "vmwarevsphere-hostsystem",
+			Usage:  "vSphere compute resource where the docker VM will be instantiated. This can be omitted if using a cluster with DRS",
+		},
+		mcnflag.StringSliceFlag{
+			EnvVar: "VSPHERE_CFGPARAM",
+			Name:   "vmwarevsphere-cfgparam",
+			Usage:  "vSphere vm configuration parameters (used for guestinfo)",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_CLOUDINIT",
+			Name:   "vmwarevsphere-cloudinit",
+			Usage:  "vSphere cloud-init filepath or url to add to guestinfo, filepath will be read and base64 encoded before adding",
+		},
+		mcnflag.StringFlag{
+			EnvVar: "VSPHERE_CLOUD_CONFIG",
+			Name:   "vmwarevsphere-cloud-config",
+			Usage:  "Filepath to a cloud-config yaml file to put into the ISO user-data",
+		},
```

Comment:
1. The method used to inject the cloud config (guestinfo or ISO userdata) depends on whether VMs are cloned or live booted from ISO. Is it still necessary to have two different flags to provide cloud-config files?

2. I don't think Rancher allow users to upload a cloud-config file. What is the use case for being able to specfiy a filepath?  I think we should accept base64 encoded cloud config as inline parameter, so that it can be part of the node template.

Reply:
1 i wanted to leave the old path to rancheros intact as much as possible and added new fields to supplement the new ideas. youre suggesting we combine cloud-config and cloud-init as one flag?
2  we have code to convert a textarea to a file and pass it in and on the cmd line it's easier to use the filepath in my experience.